libqtile/widget/mpris2widget.py

```diff
@@ -22,158 +22,154 @@
-    is displayed is configurable.
+    a reasonably correct version of MPRIS. This widget scrolls the text if
+    necessary and information that is displayed is configurable.
     """
-    orientations = base.ORIENTATION_HORIZONTAL
-    defaults = [
-        ('name', 'audacious', 'Name of the MPRIS widget.'),
 
+    defaults = [
         ('objname', 'org.mpris.MediaPlayer2.audacious',
             'DBUS MPRIS 2 compatible player identifier'
             '- Find it out with dbus-monitor - '
             'Also see: http://specifications.freedesktop.org/'
             'mpris-spec/latest/#Bus-Name-Policy'),
-
-        ('display_metadata', ['xesam:title', 'xesam:album', 'xesam:artist'],
-            'Which metadata identifiers to display. '
-            'See http://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/#index5h3 '
-            'for available values'),
-
-        ('scroll_chars', 30, 'How many chars at once to display.'),
-        ('scroll_interval', 0.5, 'Scroll delay interval.'),
-        ('scroll_wait_intervals', 8, 'Wait x scroll_interval before'
-            'scrolling/removing text'),
-
-        ('stop_pause_text', None, "Optional text to display when in the stopped/paused state"),
+        ('format', '{play_status}  <b>{title}</b> - {artist}',
+            'How to format the status metadata.'
+            'Accepted keys are: "play_status" and Xesam properties.'
+            'https://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/'),
+        ('play_states', play_states, 'dict of `play_status` strings.'
+            'Accepted keys are "play" and "pause".'),
+        ('idle_message', '', 'Text to display when the player is idle'),
+        ('enable_scrolling', True, 'Wether to scroll the text.'),
     ]
 
     def __init__(self, **config):
-        base._TextBox.__init__(self, '', **config)
+        base._ScrollText.__init__(self, **config)
         self.add_defaults(Mpris2.defaults)
 
-        self.scrolltext = None
-        self.displaytext = ''
-        self.is_playing = False
-        self.scroll_timer = None
-        self.scroll_counter = None
-        self.count = 0
+        self.player = None
+        self.add_callbacks({
+            'Button1': self.send_cmd('play_pause'),
+            'Button4': self.send_cmd('next'),
+            'Button5': self.send_cmd('previous'),
+        })
+
+        self.metadata = {}
+        self.play_status = ''
+        self.is_playing = None
+        self.text = self.idle_message
 
     async def _config_async(self):
         subscribe = await add_signal_receiver(
-                        self.message,
-                        session_bus=True,
-                        signal_name="PropertiesChanged",
-                        bus_name=self.objname,
-                        path="/org/mpris/MediaPlayer2",
-                        dbus_interface="org.freedesktop.DBus.Properties")
+            self.message,
```

Comment:
This looks like a bunch of whitespace changes; can we drop these whitespace changes (and all others, e.g. the `"".format() -> f""`) from this patch?